bundles/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/discovery/DSMRMeterDiscoveryService.java

```diff
@@ -148,8 +144,9 @@ private void validateConfiguredMeters(List<Thing> things, Set<DSMRMeterType> con
                 .filter(DSMRMeterHandler.class::isInstance)
                 .map(DSMRMeterHandler.class::cast)
                 .map(h -> h == null ? null : h.getMeterDescriptor())
-                .map(d -> d == null ? null : d.getMeterType())
-                .filter(Objects::nonNull)
+                .map(d -> Optional.ofNullable(d == null ? null : d.getMeterType()))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
```

Comment:
What was wrong with this code that required this change?

Reply:
The null-checker is not able to detect that a `Stream<@Nullable T>` is convferted to a `Stream<@NonNull T>` if `.filter(Objects::nonNull)` is applied.